plotly/offline/offline.py

```diff
@@ -204,11 +213,30 @@ def _plot_html(figure_or_data, show_link, link_text, validate,
         link_text = link_text.replace('plot.ly', link_domain)
         config['linkText'] = link_text
 
-    script = 'Plotly.newPlot("{id}", {data}, {layout}, {config})'.format(
-        id=plotdivid,
-        data=jdata,
-        layout=jlayout,
-        config=jconfig)
+    if 'frames' in figure_or_data:
+        script = '''
+        Plotly.plot(
+            '{id}',
+            {data},
+            {layout},
+            {config}
+        ).then(function () {add_frames}).then(function(){animate})
```

Comment:
See: https://github.com/plotly/plotly.js/pull/1054

That PR has been merged and makes the syntax somewhat simpler. I'd elaborate here, but the PR description explains it fairly clearly. Calling `addFrames` separately will remain perfectly fine, but ideally will become unnecessary.